framework/src/media/utils/MediaUtils.cpp

```diff
@@ -1002,23 +1028,19 @@ bool file_header_parsing(FILE *fp, audio_type_t audioType, unsigned int *channel
 	unsigned char tag[2];
 	bool isHeader;
 	int ret;
+	bool res;
 
 	switch (audioType) {
 #ifdef CONFIG_CODEC_MP3
 	case AUDIO_TYPE_MP3:
 		isHeader = false;
 		/* https://id3.org/d3v2.3.0 check id3v3 tag in file */
-		unsigned char id3Header[ID3_HEADER_LENGTH];
-		ret = fread(id3Header, sizeof(unsigned char), ID3_HEADER_LENGTH, fp);
-		if (strncasecmp((const char *)id3Header, "ID3", 3) == 0) {
-			medvdbg("string id3v2 tag found\n");
-			/* todo: check for extended header and parse title, artist etc details */
-			fseek(fp, 128, SEEK_SET);
-			break;
-		} else {
-			medvdbg("string id3v2 tag not found\n");
-			fseek(fp, 0, SEEK_SET);
-			break;
+		unsigned char id3Header[MP3_ID3V2_HEADER_LENGTH];
+		ret = fread(id3Header, sizeof(unsigned char), MP3_ID3V2_HEADER_LENGTH, fp);
```

Comment:
lets perform this fread operation also inside checkId3V2Header header. we have passed isFile (true/false) param.